core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -210,6 +211,21 @@ private static void toJson(TableMetadata metadata, JsonGenerator generator) thro
     }
     generator.writeEndObject();
 
+    writeSnapshotRelated(metadata, generator);
+
+    generator.writeArrayFieldStart(METADATA_LOG);
+    for (MetadataLogEntry logEntry : metadata.previousFiles()) {
+      generator.writeStartObject();
+      generator.writeNumberField(TIMESTAMP_MS, logEntry.timestampMillis());
+      generator.writeStringField(METADATA_FILE, logEntry.file());
+      generator.writeEndObject();
+    }
+    generator.writeEndArray();
+
+    generator.writeEndObject();
+  }
+
+  private static void writeSnapshotRelated(TableMetadata metadata, JsonGenerator generator) throws IOException {
```

Comment:
It doesn't look like this refactor is necessary any more. It just creates a method that is only used once. I don't think that this file needs to change at all.